app/playwright.config.ts

```diff
@@ -0,0 +1,83 @@
+import process from "node:process";
+import { defineConfig, devices } from "@playwright/test";
+import path from "path";
+
+const setupProject = { name: "setup", testMatch: /.*\.setup\.ts/ };
+
+const outputDir = "test-results";
+const authStorageStateFile = path.join(__dirname, outputDir, `auth.setup.ts-authenticate-setup/auth.json`);
+const baseURL = process.env.PACKIT_E2E_BASE_URL || "http://localhost:3000";
+const testMatch = baseURL.startsWith("http://localhost") ? /.*\.(local|spec)\.ts/ : /.*\.spec\.ts/;
```

Comment:
I think this should be an environment variable (possibly with a fall back based on the URL).
We can arrange for an instance on `packit-dev` to mirror the contents of this repository, in which case we'll be able to run those tests there too.

Reply:
Alternatively we could use tags for this: https://playwright.dev/docs/test-annotations#tag-tests

This has two benefits: it allows us to specify whether or not to enable them on the command line instead of guessing based on the hostname, and it means we don't have to partition our tests into different files (especially when we introduce tests that cause mutations, which will add yet another axis along which tests are partitioned).

Reply:
Tags are fine but I'm wary of accidentally running tests against servers they shouldn't be run on. so I think we'd still want some sort of sanity test that the base url looks right for the test suite, probably in the suite itself. 
I think the generic vs specific tests do fall naturally into different suites so I don't mind that too much, but yeah, mutation tests will be another dimension again.

Reply:
Actually, it would probably make sense to put this sanity check in auth.setup.ts since this is going to run for every test which needs to authenticate. It has access to the `testInfo` for the test, which includes its tags, so it can check if we're running a test  with the `@mutatesState` tag against non-localhost or non-dev server and bomb out if so.